infra/testApi.mjs

```diff
@@ -3,176 +3,129 @@ import { strict as assert } from 'node:assert';  // use strict equality everywhe
-  await new Promise(r => setTimeout(r, 100)); // ms
+// MARK: Analyze
+const expected_analyze = [[[14.97651307489794], "2.3"]]
+const analyze_result1 = await callAnalyze(FPCoreFormula, [[[
+  14.97651307489794
+], 0.12711304680349078]], false);
+
+const analyze_result2 = await callAnalyze(FPCoreFormula, [[[
+  14.97651307489794
+], 0.12711304680349078]], true);
+
+assert.deepEqual(analyze_result1.points, expected_analyze);
+assert.deepEqual(analyze_result2.points, expected_analyze);
+
+async function callAnalyze(fpcore, p_context, async_huh) {
+  const analyze_json = await getJSONFor('analyze', 'POST', JSON.stringify({
+    formula: fpcore, sample: p_context
+  }), async_huh);
+  assertIdAndPath(analyze_json);
+  checkFelids(analyze_json, ['command', 'points', 'job', 'path']);
+  assert.equal(analyze_json.points.length == p_context.length, true);
+  return analyze_json
 }
-assert.equal(checkStatus.statusText, 'Job complete')
 
-// up endpoint
-const up = await fetch(makeEndpoint("/up"), { method: 'GET' })
-assert.equal('Up', up.statusText) // Herbie runs single thread on CI.
+// Test with a likely missing job-id
+const badTimelineRSP = await fetchAndCheckRSPHeaders("/timeline/42069", { method: 'GET' }, CHECK_CORS)
+assert.equal(badTimelineRSP.status, 404);
 
-// Sample endpoint
-const sampleBody = {
-  method: 'POST',
-  body: JSON.stringify({ formula: FPCoreFormula2, seed: 5 })
+// MARK: Check Status
+const check_missing_job = await fetchAndCheckRSPHeaders(`/check-status/42069`, { method: 'GET' }, CHECK_CORS)
+assert.equal(check_missing_job.status, 202);
+
+// MARK: UP
+async function callUp() {
+  const up = await fetchAndCheckRSPHeaders("/up", { method: 'GET' }, CHECK_CORS);
+  assert.equal(up.statusText, 'Up');
 }
-const sampleRSP = await fetch(makeEndpoint("/api/sample"), sampleBody)
-const sampleAsyncResult = await callAsyncAndWaitJSONResult("/api/start/sample", sampleBody)
-const jid = sampleRSP.headers.get("x-herbie-job-id")
-assert.notEqual(jid, null)
-const sample = await sampleRSP.json()
-assertIdAndPath(sampleAsyncResult)
-assert.ok(sampleAsyncResult.points)
-assert.equal(sampleAsyncResult.points.length, SAMPLE_SIZE)
-assertIdAndPath(sample)
-assert.ok(sample.points)
-assert.equal(sample.points.length, SAMPLE_SIZE, `sample size should be ${SAMPLE_SIZE}`)
-
-// Make second call to test that results are the same
-const sample2RPS = await fetch(makeEndpoint("/api/sample"), sampleBody)
-const jid2 = sample2RPS.headers.get("x-herbie-job-id")
-assert.notEqual(jid2, null)
-const sample2 = await sample2RPS.json()
-assertIdAndPath(sample2)
-assert.deepEqual(sample.points[1], sample2.points[1])
-
-//Explanations endpoint
-const explainBody = {
-  method: 'POST',
-  body: JSON.stringify({
-    formula: FPCoreFormula, sample: sample.points
-  })
+
+// MARK: Sample
+async function callSample(fpcore, async_huh) {
+  var sample_json = await getJSONFor('sample', 'POST', JSON.stringify({ formula: fpcore, seed: 5 }), async_huh);
```

Comment:
As far as I can tell, `getJSONFor` is always called with a method of `POST` and the body is always JSON. Can we move that into `getJSONFor`?